### ISSUE TITLE
Take callback name as command when not provided

### ISSUE THREAD

**Issue by aditya-yadav-27** [python-telegram-bot/python-telegram-bot#2844]
### What kind of feature are you missing? Where do you notice a shortcoming of PTB?

When we write **Command/Prefix handlers** we define the commands every time (even if the name of callback and command is same everytime) which is not a good practice ofcourse

### Describe the solution you'd like

It will be better that, if we don't provide command it will automatically take the name of callback that is provided for e.g.
`def start(update, context):`
 `....this is start`
​​`dp.add_handler(CommandHandler(start))`
this should listen to `/start`
it will help if you are writing many handlers and the name of callbacks is equal to the command you want so you don't want to write the command everytime.

**Comment by harshil21** [comment]
Sounds like a good idea! Most people have the callback name as command name anyway (except for `help` since that's a built-in python func)

**Comment by Bibo-Joshi** [comment]
Somehow missed this :D Thanks for reaching out! I'm not sure about the idea though. For three reasons.

---

Firstly, this doesn't generalize well. `functools.partial(…)` doesn't have a `__name__` and neither does something like `CallableClassInstance()`. These *can* be handled, since accessing `__name__` will raise an `AttributeError`. Decorated functions where the decorator doesn't use `functools.wraps` are harder, because `__name__` is available but not the name that one would expect. PTB would not be able to catch that, the command wouldn't work as expected and this would be hard to debug. So the user would need to be quite aware what they're doing and it feels like we'd be making a specific coding pattern implicitly the expected default.

Secondly, it wolud make the usage of `CommandHandler` more complicated. We currently have an optional parameter `filters`, so the signature would have to be changed to either something like

```python
def __init__(self, command: str | List[str] | Callable, callback: Callable | BaseFilter = None, filters: BaseFilter = None)
```
or something like

```python
def __init__(self, callback: Callable, (*, )command: str | List[str]=None, filters: BaseFilter = None)
```
The first means a number of `if-else` causes in `__init__` to check if the input is acceptable and to properly integrate with static type checking, we'd need some `@overloads`.
The second is breaking, which maybe isn't a big problem for v14, but it feels a bit unnecessary to me.

Thirdly: maintenance work. I have a gut feeling that this is the kind of feature where we'd have to add special handling for another special case every now and then, making `CommandHandler.__init__`  blow up. + Answering questions from users that don't understand the limitations of the feature.

---
In summary, I'm hesitant to change the current behavior in the main lib. Can you understand my arguments?
Would a contribution to ptbcontrib be an option instead?

**Comment by aditya-yadav-27** [comment]
Forgot to mention, I got this idea from [discord.py](https://github.com/Rapptz/discord.py)
